DDG4/edm4hep/EDM4hepFileReader.cpp

```diff
@@ -0,0 +1,320 @@
+//==========================================================================
+//  AIDA Detector description implementation
+//--------------------------------------------------------------------------
+// Copyright (C) Organisation europeenne pour la Recherche nucleaire (CERN)
+// All rights reserved.
+//
+// For the licensing terms see $DD4hepINSTALL/LICENSE.
+// For the list of contributors see $DD4hepINSTALL/doc/CREDITS.
+//
+// Author     : A.Sailer
+//
+//==========================================================================
+
+/** \addtogroup Geant4EventReader
+ *
+ @{
+ \package EDM4hepFileReader
+ * \brief Plugin to read EDM4hep files
+ *
+ *
+ @}
+*/
+
+#include <CLHEP/Units/SystemOfUnits.h>
+
+#include <DDG4/EventParameters.h>
+#include <DDG4/Factories.h>
+#include <DDG4/Geant4InputAction.h>
+#include <DDG4/RunParameters.h>
+
+#include <edm4hep/EventHeaderCollection.h>
+#include <edm4hep/MCParticleCollection.h>
+
+#include <podio/Frame.h>
+#include <podio/ROOTReader.h>
+
+typedef dd4hep::detail::ReferenceBitMask<int> PropertyMask;
+
+
+namespace dd4hep::sim {
+
+  // we use the index of the objectID to identify particles
+  // we will not support MCParticles from different collections
```

Comment:
I suppose the same constraint is also in place for LCIO (or in general for various readers)? I think it's pretty reasonable, but want to make sure that this is documented somwhere (maybe it already is).

Reply:
Since once can only set one MCParticle collection name at the moment, this is implicitly the case. It would be a feature request to read more than one MCParticle collection.

Reply:
Ah right, in that case I think this is "documented" enough.

Reply:
This comment here was really for implementation reasons, since podio::ObjectID cannot be used as a map key due to missing operator< (<=?) so I went the lazy route.